libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -336,6 +337,68 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     return crypto.createHash('sha1').update(shrinkwrapContent).digest('hex');
   }
 
+  /**
+   * Determine whether `pnpm-lock.yaml` contains insecure sha1 hashes.
+   * @internal
+   */
+  private _disallowInsecureSha1(
+    customTipsConfiguration: CustomTipsConfiguration,
+    terminal: ITerminal
+  ): boolean {
+    const { packages } = this._shrinkwrapJson;
+    if (packages) {
+      for (const { resolution } of Object.values(packages)) {
+        if ((resolution?.integrity.startsWith('sha1')) {
+          terminal.writeErrorLine(
+            'Error: An integrity field with "sha1" was found in pnpm-lock.yaml;' +
+              ' this conflicts with the "disallowInsecureSha1" policy from pnpm-config.json.\n'
+          );
+
+          customTipsConfiguration._showErrorTip(terminal, CustomTipId.TIP_PNPM_DISALLOW_INSECURE_SHA1);
+
+          return true; // Indicates an error was found
+        }
+      }
+    }
+    return false;
+  }
+
+  /** @override */
+  public validateShrinkwrapAfterUpdate(rushConfiguration: RushConfiguration, terminal: ITerminal): void {
+    const pnpmLockfilePolicies: [string, boolean][] = Object.entries(
+      rushConfiguration.pnpmOptions.pnpmLockfilePolicies ?? {}
+    );
+
+    if (pnpmLockfilePolicies && pnpmLockfilePolicies.length > 0) {
+      let invalidPoliciesCount: number = 0;
+
+      for (const [policy, enabled] of pnpmLockfilePolicies) {
+        if (enabled) {
+          switch (policy) {
+            case 'disallowInsecureSha1': {
+              const isError: boolean = this._disallowInsecureSha1(
+                rushConfiguration.customTipsConfiguration,
+                terminal
+              );
+              if (isError) {
+                invalidPoliciesCount += 1;
+              }
+              break;
+            }
+
+            default: {
+              throw new Error(`Unknown pnpm lockfile policy "${policy}"`);
+            }
+          }
+        }
+      }
```

Comment:
This is an awkward way to go through an object with a well-known type. This is easier to read:

```suggestion
      if (pnpmLockfilePolicies.disallowInsecureSha1) {
        const isError: boolean = this._disallowInsecureSha1(
          rushConfiguration.customTipsConfiguration,
          terminal
        );
        if (isError) {
          invalidPoliciesCount++;
        }
      }
```

And then as more policies are added, they'll get included as their own `if` statements after this one.